beacon/sync/src/main/java/tech/pegasys/teku/beacon/sync/forward/multipeer/batches/SyncSourceBatch.java

```diff
@@ -299,6 +355,76 @@ public String toString() {
         .toString();
   }
 
+  private boolean validateNewBlocks(final List<SignedBeaconBlock> newBlocks) {
+    if (blocks.isEmpty() || newBlocks.isEmpty()) {
+      return true;
+    }
+    final SignedBeaconBlock previousBlock = blocks.get(blocks.size() - 1);
+    final SignedBeaconBlock firstNewBlock = newBlocks.get(0);
+    if (!firstNewBlock.getParentRoot().equals(previousBlock.getRoot())) {
+      LOG.debug(
+          "Marking batch invalid because new blocks do not form a chain with previous blocks");
+      return false;
+    }
+    return true;
+  }
+
+  private boolean validateNewBlobSidecars(
+      final SyncSource syncSource,
+      final List<SignedBeaconBlock> newBlocks,
+      final Map<Bytes32, List<BlobSidecar>> newBlobSidecarsByBlockRoot) {
+    final Set<Bytes32> blockRootsWithKzgCommitments = new HashSet<>(newBlocks.size());
+    for (final SignedBeaconBlock block : newBlocks) {
+      final Bytes32 blockRoot = block.getRoot();
+      final List<BlobSidecar> blobSidecars =
+          newBlobSidecarsByBlockRoot.getOrDefault(blockRoot, List.of());
+      final int numberOfKzgCommitments =
+          block
+              .getMessage()
+              .getBody()
+              .toVersionDeneb()
+              .map(BeaconBlockBodyDeneb::getBlobKzgCommitments)
+              .map(SszList::size)
+              .orElse(0);
+      if (numberOfKzgCommitments > 0) {
+        blockRootsWithKzgCommitments.add(blockRoot);
+      }
+      if (blobSidecars.size() != numberOfKzgCommitments) {
+        LOG.debug(
+            "Marking batch invalid because {} blob sidecars were received, but the number of KZG commitments in a block ({}) were {}",
+            blobSidecars.size(),
+            blockRoot,
+            numberOfKzgCommitments);
+        return false;
+      }
+      final UInt64 blockSlot = block.getSlot();
+      for (final BlobSidecar blobSidecar : blobSidecars) {
+        if (!blobSidecar.getSlot().equals(blockSlot)) {
+          LOG.debug(
+              "Marking batch invalid because blob sidecar for root {} was received with slot {} which is different than the block slot {}",
+              blockRoot,
+              blobSidecar.getSlot(),
+              blockSlot);
+          return false;
+        }
+      }
+    }
+
+    final SetView<Bytes32> unexpectedBlobSidecarsRoots =
+        Sets.difference(newBlobSidecarsByBlockRoot.keySet(), blockRootsWithKzgCommitments);
+    if (!unexpectedBlobSidecarsRoots.isEmpty()) {
+      LOG.debug(
+          "Applying minor penalty to peer {} because unexpected blob sidecars with roots {} were received",
+          syncSource,
+          unexpectedBlobSidecarsRoots);
+      syncSource.adjustReputation(ReputationAdjustment.SMALL_PENALTY);
```

Comment:
to follow the design, shouldn't we have it inside the conflictResolutionStrategy?

Reply:
Yeah, but reportInvalidBatch results in a LARGE_PENALTY for the peer, is that what we would like in this scenario?

Reply:
I see like another method exposed there to cover this specific situation..

Reply:
Kk, did a change, not 100% sure about naming, but think it makes sense.